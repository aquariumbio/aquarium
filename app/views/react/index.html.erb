<div id="react-content" class="react_container"></div>
<script>
      'use strict';

      const createElement = React.createElement;


      function AssignedJobTitle(props) {
        return (
          createElement('div', { className: 'assigned_job_table_title fflex_resp fflex_title' },
            createElement('div', { className: 'job-tecnician-name fflex_col' },'Technician'),
            createElement('div', { className: 'job-id fflex_col' }, 'Job'),
            createElement('div', { className: 'job-protocol fflex_col2' },'Protocol'),
            createElement('div', { className: 'job-status fflex_col' }, 'Status'),
            createElement('div', { className: 'job-operations fflex_col fflex_center_text' }, 'Operations'),
            createElement('div', { className: 'job-assigned-time fflex_col fflex_center_text' }, 'Assigned'),
            createElement('div', { className: 'job-start-time fflex_col fflex_center_text' }, 'Started'),
          )
        )
      }

      function AssignedJobRow(props) {
        return (
          createElement('div', { className: 'job fflex_resp fflex_row' },
            createElement('div', { className: 'job-tecnician-name fflex_col' }, props.job.to_name || '-'),
            createElement('div', { className: 'job-id fflex_col' },
              createElement('a', { href: `/jobs/${props.job.id}`, className: 'google_blue' }, props.job.id)
            ),
            createElement('div', { className: 'job-protocol fflex_col2' },
              createElement('a', { className: 'google_blue', href: `/jobs/${props.job.id}` }, props.job.operation_type_name)
            ),
            createElement('div', { className: 'job-status fflex_col' }, props.job.pc == 0 ? 'Running' : 'Ready'),
            createElement('div', { className: 'job-operations fflex_col fflex_center_text' }, props.job.operation_count),
            createElement('div', { className: 'job-assigned-time fflex_col fflex_center_text' }, props.job.assigend_at ? new Date(props.job.assigend_at).toLocaleDateString('en-US') : '-'),
            createElement('div', { className: 'job-start-time fflex_col fflex_center_text' }, props.job.pc == 0 ? 'TODO parse start' : '-'),
          )
        )
      }
    

// 
      class AssignedJobsTable extends React.Component {
        constructor(props) {
          super(props);
          this.state = { 
            isLoading: true,
            hasError: false,
            liked: false 
          };

          this.getManagerDashboard = this.getManagerDashboard.bind(this)

        }

        componentDidMount() {
            this.getManagerDashboard()

            let interval = 300000 // 5 minutes

            this.fetch = setInterval(() => {
              this.getManagerDashboard()
            }, interval); 
        }

        componentWillUnmount() {
          clearInterval(this.fetch);
        }

        static getDerivedStateFromError(error) {
          // Update state so the next render will show the fallback UI.
          return { hasError: true };
        }

        componentDidCatch(error, errorInfo) {
          // You can also log the error to an error reporting service
          console.log(error, errorInfo);
        }

        render() {

          let title = createElement('div', {className: 'title'}, 'Active Jobs');

          if (this.state.hasError) {
            // You can render any custom fallback UI
            return createElement('div', {}, "Something went wrong.")
          }

          let content
          if (this.state.isLoading) {
            content = createElement(
              'p',
              {},
              'Loading...',
            )
          }
          
          if (!this.state.isLoading && !this.state.hasError){      

            content = this.state.jobs.map(function(job, i) {
                return createElement( AssignedJobRow, {key: i + 1, job: job})
              }
            )

            content.unshift(createElement(AssignedJobTitle, {key: 0}))
          }

          return createElement('div', {className: 'fflex_768'}, title, content);
        }
      
        getManagerDashboard() {
          fetch('/api/v2/dashboard')
          .then(result => result.json())
          .then(
            (result) => {
              this.setState({
                isLoading: false,
                jobs: result.data
              });
              console.log("response");
              console.log(result.data);
            },
            (error) => {
              console.log("error:" + JSON.stringify(error));
              this.setState({
                isLoading: false,
                error
              });
            }
          )
        }
        
      }

          

      const domContainer = document.querySelector('#react-content');
      ReactDOM.render(createElement(AssignedJobsTable), domContainer);
</script>
