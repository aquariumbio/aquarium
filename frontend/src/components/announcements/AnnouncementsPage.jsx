/* eslint-disable */
import React, { useState, useEffect } from 'react';
import Paper from '@material-ui/core/Paper';
import announcementsAPI from '../../helpers/api/announcementsAPI';
import { makeStyles } from '@material-ui/core/styles';
import {
  Grid,
} from '@material-ui/core';
import CreateAnnouncementDialog from './CreateAnnouncementDialog';
import AnnouncementsTable from './AnnouncementsTable';

const useStyles = makeStyles((theme) => ({
  root: {
    height: '100%'
  },
  paper: {
    minHeight: '90vh'
  },
  dialogGrid: {
    position: 'sticky',
    height: '100%',
    left: theme.spacing(0),
    top: theme.spacing(0)
  },
  table: {
    height: '100%'
  }
}));

const AnnouncementsPage = () => {
  const classes = useStyles();
  const [rowData, setRowData] = useState([]);
  const [title, setTitle] = useState(null);
  const [message, setMessage] = useState(null);
  const [active, setActive] = useState(false);

  useEffect(() => {
    const handleGetAnnouncements = async () => {
      var announcementsData = await announcementsAPI.getAllAnnouncements();
      setRowData(announcementsData)
    }

    handleGetAnnouncements();
  }, []);

  const handleCreateAnnouncement = async () => {
    var response = await announcementsAPI.createAnnouncement(
      {
        "announcement": {
          "title": title,
          "message": message,
          "active": active
        }
      }
    )
    setRowData((rowData) => [
      ...rowData,
      {
        id: rowData.length + 1, title: title, message: message, active: active,
      },
    ]);
  };

  const handleTitle = (event) => {
    setTitle(event.target.value);
  };

  const handleMessage = (event) => {
    setMessage(event.target.value);
  };

  const handleActive = (event) => {
    setActive(event.target.checked);
  };

  const handleDelete = async (event, row) => {
    confirm('You want to delete ' + row.title)
    console.log(rowData.id)
    var announcementsData = await announcementsAPI.deleteAnnouncement(row.id);
    //setRowData(announcementsData)
    setRowData(rowData.filter(item => item.id !== rowData.id));
  }


  return (
    <>
      <Paper elevation={3} className={classes.paper}>
        <Grid
          container
          direction="row">
          <Grid item xs={3} sm={3} className={classes.dialogGrid}>
            <CreateAnnouncementDialog
              handleCreateAnnouncement={handleCreateAnnouncement}
              handleTitle={handleTitle}
              handleMessage={handleMessage}
              handleActive={handleActive}
              active={active}
            />
          </Grid>
          <Grid item xs={12} sm={9} className={classes.table}>
            <AnnouncementsTable
              rowData={rowData}
              setRowData={setRowData}
            />
          </Grid>
        </Grid>
      </Paper>
    </>
  );
};

export default AnnouncementsPage;
