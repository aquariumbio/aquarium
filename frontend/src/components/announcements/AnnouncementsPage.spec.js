import React from 'react';
import { render } from "@testing-library/react";
import AnnouncementsPage from './index.jsx';

describe("Describes Announcements page", () => {
    it("It renders without crashing", () => {
        const { asFragment } = render(
            <AnnouncementsPage />
        );
        const firstRender = asFragment()

        //debug();
        expect(firstRender).toMatchSnapshot();
    }
    )

});
