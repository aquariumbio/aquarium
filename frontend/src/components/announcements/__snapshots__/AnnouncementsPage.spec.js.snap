// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Describes Announcements page It renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation3 MuiPaper-rounded"
    data-testid="1"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        Create Announcement
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded"
      style="position: relative;"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MTableToolbar-root-11 MuiToolbar-gutters"
      >
        <div
          class="MTableToolbar-title-15"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
            style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
          >
            Announcements
          </h6>
        </div>
        <div
          class="MTableToolbar-spacer-13"
        />
        <div
          class="MuiFormControl-root MuiTextField-root MTableToolbar-searchField-16"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                title="Search"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              aria-label="Search"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
              placeholder="Search"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="Clear Search"
                class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    aria-label="clear"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="MTableToolbar-actions-14"
        >
          <div>
            <div>
              <span>
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
                  tabindex="0"
                  title="Add"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="Component-horizontalScrollContainer-18"
        style="overflow-x: auto; position: relative;"
      >
        <div>
          <div
            style="overflow-y: auto;"
          >
            <div>
              <table
                class="MuiTable-root"
                style="table-layout: auto;"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-19 MuiTableCell-alignLeft"
                      scope="col"
                      style="box-sizing: border-box;"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root"
                        role="button"
                        tabindex="0"
                      >
                        <div
                          aria-describedby="rbd-hidden-text-0-hidden-text-0"
                          data-rbd-drag-handle-context-id="0"
                          data-rbd-drag-handle-draggable-id="0"
                          data-rbd-draggable-context-id="0"
                          data-rbd-draggable-id="0"
                          draggable="false"
                          role="button"
                          tabindex="0"
                        >
                          Id
                        </div>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                          />
                        </svg>
                      </span>
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-19 MuiTableCell-alignLeft"
                      scope="col"
                      style="box-sizing: border-box;"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root"
                        role="button"
                        tabindex="0"
                      >
                        <div
                          aria-describedby="rbd-hidden-text-0-hidden-text-0"
                          data-rbd-drag-handle-context-id="0"
                          data-rbd-drag-handle-draggable-id="1"
                          data-rbd-draggable-context-id="0"
                          data-rbd-draggable-id="1"
                          draggable="false"
                          role="button"
                          tabindex="0"
                        >
                          Title
                        </div>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                          />
                        </svg>
                      </span>
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-19 MuiTableCell-alignLeft"
                      scope="col"
                      style="box-sizing: border-box;"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root"
                        role="button"
                        tabindex="0"
                      >
                        <div
                          aria-describedby="rbd-hidden-text-0-hidden-text-0"
                          data-rbd-drag-handle-context-id="0"
                          data-rbd-drag-handle-draggable-id="2"
                          data-rbd-draggable-context-id="0"
                          data-rbd-draggable-id="2"
                          draggable="false"
                          role="button"
                          tabindex="0"
                        >
                          Message
                        </div>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                          />
                        </svg>
                      </span>
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-19 MuiTableCell-alignLeft"
                      scope="col"
                      style="box-sizing: border-box;"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root"
                        role="button"
                        tabindex="0"
                      >
                        <div
                          aria-describedby="rbd-hidden-text-0-hidden-text-0"
                          data-rbd-drag-handle-context-id="0"
                          data-rbd-drag-handle-draggable-id="3"
                          data-rbd-draggable-context-id="0"
                          data-rbd-draggable-id="3"
                          draggable="false"
                          role="button"
                          tabindex="0"
                        >
                          Active
                        </div>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                          />
                        </svg>
                      </span>
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-19 MuiTableCell-paddingCheckbox"
                      scope="col"
                      style="width: 96px; text-align: center; box-sizing: border-box;"
                    >
                      <span
                        aria-disabled="true"
                        class="MuiButtonBase-root MuiTableSortLabel-root Mui-disabled"
                        role="button"
                        tabindex="-1"
                      >
                        Actions
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root"
                    style="height: 245px;"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                      colspan="5"
                      style="padding-top: 0px; padding-bottom: 0px; text-align: center;"
                    >
                      No records to display
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
      <table
        class="MuiTable-root"
      >
        <tfoot
          class="MuiTableFooter-root"
          style="display: grid;"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-footer"
          >
            <td
              class="MuiTableCell-root MuiTableCell-footer MuiTablePagination-root Component-paginationRoot-7"
              colspan="3"
              icons="[object Object]"
              style="float: right; overflow-x: auto;"
            >
              <div
                class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar Component-paginationToolbar-8 MuiToolbar-gutters"
              >
                <div
                  class="MuiTablePagination-spacer"
                />
                <p
                  class="MuiTypography-root MuiTablePagination-caption Component-paginationCaption-9 MuiTypography-body2 MuiTypography-colorInherit"
                  id="mui-19822"
                >
                  Rows per page:
                </p>
                <div
                  class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot Component-paginationSelectRoot-10"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="mui-19822 mui-76697"
                    class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
                    id="mui-76697"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      style="padding: 0px 5px;"
                    >
                      5 rows 
                    </div>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value="5"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
                <p
                  class="MuiTypography-root MuiTablePagination-caption Component-paginationCaption-9 MuiTypography-body2 MuiTypography-colorInherit"
                >
                  0-0 of 0
                </p>
                <div
                  class="MTablePaginationInner-root-20"
                >
                  <span
                    class=""
                    title="First Page"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                          />
                        </svg>
                      </span>
                    </button>
                  </span>
                  <span
                    class=""
                    title="Previous Page"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                          />
                        </svg>
                      </span>
                    </button>
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-caption"
                    style="flex: 1; text-align: center; align-self: center; flex-basis: inherit;"
                  >
                    0-0 of 0
                  </span>
                  <span
                    class=""
                    title="Next Page"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                          />
                        </svg>
                      </span>
                    </button>
                  </span>
                  <span
                    class=""
                    title="Last Page"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                          />
                        </svg>
                      </span>
                    </button>
                  </span>
                </div>
              </div>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
